### PR TITLE
Feature/99 use form container in table editor

### PR DESCRIPTION
**Pull request by pixeljunkie**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#99
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/private/modules/MTTableEditor/views/includes/fields.inc.php [hunk 1]
```diff
@@ -35,16 +34,18 @@
                     $oField->oDefinition->sqlData['modifier'] = 'readonly';
                 }
 
-                $sTmpFormTabsContent .= '<tr class="row m-0 '.$rowClass.'">
-            <th valign="top" class="leftTD col-2"'.$rowColorStyle.'>
-              <div id="fieldname_'.TGlobal::OutHTML($oField->name).'">
+
+                // todo $rowColorStyle
```
Comment (umulmrum) [inline src/CoreBundle/private/modules/MTTableEditor/views/includes/fields.inc.php]: Is this a TODO we can already work on? Ideally we don't have TODOs left in production code.

File: src/CoreBundle/Resources/public/javascript/tableEditor.js [hunk 2]
```diff
@@ -539,11 +539,13 @@ function SaveViaAjax() {
...
-    // remove all message-classes from the fields
-    $('*[id^="fieldname_"]').removeClass();
+    // remove all message background classes from field containers.
+    $('*[id^="fieldname_"]').removeClass(function (index, className) {
+        return (className.match (/(^|\s)bg-\S+/g) || []).join(' ');
+    });
 
     var returnVal = false;
-    if (data != false && data != null) {
+    if (data !== false && data != null) {
```
Comment (umulmrum) [inline src/CoreBundle/Resources/public/javascript/tableEditor.js]: Are you confident enough that this change won't cause problems? The true/false rules in JS are at least as wonderful and straightforward as in PHP.

File: src/CoreBundle/private/modules/MTTableEditor/views/includes/fields.inc.php (hunk 1 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/modules/MTTableEditor/views/includes/fields.inc.php]: oooh... missed that and found out i missed the handling of the separator field type as well. Now included.

File: src/CoreBundle/Resources/public/javascript/tableEditor.js (hunk 2 above)
Comment (pixeljunkie) [reply src/CoreBundle/Resources/public/javascript/tableEditor.js]: tested this. works. So, yes.

**Comment by umulmrum** [comment]
Thanks @pixeljunkie

**Pull request merged**